koa-crud/test/integration/models/vendor.spec.ts

```diff
@@ -0,0 +1,237 @@
+import chai, { expect } from 'chai';
+
+import chaiAsPromised from 'chai-as-promised';
+
+import mongoose from 'mongoose';
+
+import { initializeDatabase } from '../../../src/lib/mongoose';
+
+import { Vendor } from '../../../src/lib/mongoose/models/vendor';
+
+import { VendorType } from '../../../src/types';
+
+chai.use(chaiAsPromised);
+
+describe('Vendor Models', () => {
+  before(async function () {
+    this.mock = null;
+    this.mockedId = mongoose.Types.ObjectId().toString();
+    await initializeDatabase();
+  });
+
+  describe('Creating a vendor', () => {
+    afterEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    beforeEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    describe('GIVEN correct inputs and SEAMLESS type', () => {
+      it('should be fulfilled', async () => {
+        const data = {
+          name: 'Luis Angelo Belmonte',
```

Comment:
I notice that you are not using fake libraries to generate fake output. Please use `chance.js` to create fake names.

This will prevent magic string in testing and less expensive when we are to refactor something

Reply:
noted sir